src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1422,6 +1421,12 @@ public static bool IsInstanceContext(this SyntaxTree syntaxTree, int position, S
                 return false;
             }
 
+            var expressionBody = GetExpressionBodyFromMember(containingMember);
+            if (expressionBody != null)
+            {
+                return ArrowExpressionClauseExpressionAdjustedSpan(expressionBody).IntersectsWith(position);
```

Comment:
@Pilchie Could you explain the scenario here? By adjusting the span I am making sure I cover all the legit span both in broken syntax and valid syntax

If I am using a check to look for the left token then we should some more logic to handle cases like `int M => 0 * $$`